deploy/001_deploy_contracts.ts

```diff
@@ -0,0 +1,279 @@
+    },
+    {
+      name: ContractNames.ticketingParameters,
+      args: [
+        config.TicketingParameters.faceValue,
+        config.TicketingParameters.baseLiveWinProb,
+        config.TicketingParameters.expiredWinProb,
+        config.TicketingParameters.decayRate,
+        config.TicketingParameters.ticketDuration,
+      ],
+    },
+    {
+      name: ContractNames.epochsManager,
+      args: [
+        config.Seekers,
+        contracts[ContractNames.directory].address,
+        contracts[ContractNames.registries].address,
+        contracts[ContractNames.ticketingParameters].address,
+        config.EpochsManager.epochDuration,
+      ],
+    },
+    {
+      name: ContractNames.stakingManager,
+      args: [
+        config.SyloToken,
+        contracts[ContractNames.rewardsManager].address,
+        contracts[ContractNames.epochsManager].address,
+        contracts[ContractNames.stakingManager].address,
+        config.StakingManager.minimumStakeProportion,
+      ],
+    },
+    {
+      name: ContractNames.rewardsManager,
+      args: [
+        config.SyloToken,
+        contracts[ContractNames.stakingManager].address,
+        contracts[ContractNames.epochsManager].address,
+      ],
+    },
+    {
+      name: ContractNames.directory,
+      args: [
+        contracts[ContractNames.stakingManager].address,
+        contracts[ContractNames.rewardsManager].address,
+      ],
+    },
+    {
+      name: ContractNames.syloTicketing,
+      args: [
+        config.SyloToken,
+        contracts[ContractNames.registries].address,
+        contracts[ContractNames.stakingManager].address,
+        contracts[ContractNames.directory].address,
+        contracts[ContractNames.epochsManager].address,
+        contracts[ContractNames.rewardsManager].address,
+        contracts[ContractNames.authorizedAccounts].address,
+        config.Ticketing.unlockDuration,
+      ],
+    },
+  ];
+  for (const { name, args } of initializeParams) {
+    await initializeContract(name, args, deployer.address, execute);
+    printEmptyLine();
+  }
+
+  // ADD MANAGERS
+  const addManagerParams: ContractParams[] = [
+    {
+      name: ContractNames.directory,
+      args: [contracts[ContractNames.epochsManager].address],
+    },
+    {
+      name: ContractNames.rewardsManager,
+      args: [contracts[ContractNames.ticketingParameters].address],
+    },
+    {
+      name: ContractNames.rewardsManager,
+      args: [contracts[ContractNames.stakingManager].address],
+    },
+    {
+      name: ContractNames.rewardsManager,
+      args: [contracts[ContractNames.epochsManager].address],
+    },
+  ];
+  for (const { name, args } of addManagerParams) {
+    await addManager(name, deployer.address, args[0] as string, execute);
+    printEmptyLine();
+  }
+
+  await saveContracts(deployer.address, network.name, contracts, config);
+};
+
+export default func;
+func.id = 'deploy_contractss'; // id required to prevent reexecution
```

Comment:
```suggestion
func.id = 'deploy_contracts'; // id required to prevent reexecution
```

Can you explain the need to do these more? I don't see it in the docs for hardhat-deploy

Reply:
Actually, I copied this from their template. For further explanation: 
```
The function can return true if it wishes to never be executed again. This can be useful to emulate migration scripts that are meant to be executed only once. Once such script return true (async), the id field is used to track execution and if that field is not present when the script return true, it will fails. In other words, if you want a particular deploy script to run only once, it needs to both return true (async) and have an id set.
```
Since we only want to deploy the factory script once, we should set the `id` field right?